src/data/bmg_plate.py

```diff
@@ -0,0 +1,109 @@
+import os
+import numpy as np
+import pandas as pd
+import matplotlib.pyplot as plt
+
+
+class Plate:
+    """
+    Class representing a plate with values resulting from an HTS experiment
+    """
+    def __init__(self, filepath: str) -> None:
+        """
+        :param filepath: path to the file consisting plate
+        """
+        self.filepath = filepath
+        self.barcode = filepath.split('/')[-1].split('.')[0].split('\\')[-1]
+        self.plate_array = self.parse_file()
+        self.control_statistics()
+        self.z_factor = 1 - (3*(self.std_pos+self.std_neg) /
+                             (self.mean_neg-self.mean_pos))
+
+    def parse_file(self) -> np.array:
+        """
+        Read data from txt file to np.array
+
+        :return: array with plate values
+        """
+        plate = np.zeros(shape=(16, 24))
+        with open(self.filepath) as f:
```

Comment:
I think it'd be nice to separate Plate creation from usage, e.g. so that we can create a plate just by providing the data instead of having to create a file - helps with testing and keeps responsibilities separate

in this case it'd be sufficient to extract `parse_file` to a separate function and make the class itself simpler

Reply:
makes sense, will do that!